tests/ptr.rs

```diff
@@ -53,11 +53,11 @@ fn slice() {
     unsafe {
         let slice = ParticleSlice::from_raw_parts(ptr, 2);
         assert_eq!(slice.len(), 2);
-        assert_eq!(slice.name[0], "Na");
-        assert_eq!(slice.name[1], "Zn");
+        assert_eq!(slice.index(0).name, "Na");
```

Comment:
This should still be supported with the new single lengths version though, and I think it is worth keeping both tests (`slice.index(0).name == slice.name[0]`)

My issue with the `index(xx)` version is that we loose part of the appeal of SoA, since we need to load data from all member vectors to create the reference, then to throw away most of it to only access on field (here the name)

Reply:
`slice.name[0]` is actually OK. I mixed it up with `ParticleSlice` with `ParticleVec` (which cannot do so).